pkg/vm/vm_test.go

```diff
@@ -84,6 +84,104 @@ func checkVMFailed(t *testing.T, vm *VM) {
 	assert.Equal(t, true, vm.HasFailed())
 }
 
+func TestStackLimitPUSH1Good(t *testing.T) {
+	lp := MaxStackSize * 2
+	prog := make([]byte, lp)
+	for i := 0; i < lp/2; i++ {
+		prog[i] = byte(PUSH1)
+	}
+	for i := lp / 2; i < lp; i++ {
```

Comment:
`s,i := lp / 2,,/`?

Reply:
Don't understand what do you mean tbh.

Reply:
I mean you probably already have `i == lp/2`, so you can drop the initialization part:
```go
for ; i < lp; i++ {
```

Reply:
No, because `i`'s scope is restricted by first loop. Anyway, I don't see how this change will improve readability or anything.

Reply:
Ouch. True.